scipy/interpolate/_bsplines.py

```diff
@@ -1217,6 +1217,49 @@ def _make_interp_per_full_matr(x, y, t, k):
     return c
 
 
+def _handle_lhs_derivatives(t, k, xval, ab, kl, ku, deriv_ords, offset=0):
+    """ Fill in the entries of the collocation matrix corresponding to known
+    derivatives at `xval`.
+
+    The collocation matrix is in the banded storage, as prepared by _colloc.
+    No error checking.
+
+    Parameters
+    ----------
+    t : ndarray, shape (nt + k + 1,)
+        knots
+    k : integer
+        B-spline order
+    xval : float
+        The value at which to evaluate the derivatives at.
+    ab : ndarray, shape(2*kl + ku + 1, nt), Fortran order
+        B-spline collocation matrix.
+        This argument is modified *in-place*.
+    kl : integer
+        Number of lower diagonals of ab.
+    ku : integer
+        Number of upper diagonals of ab.
+    deriv_ords : 1D ndarray
+        Orders of derivatives known at xval
+    offset : integer, optional
+        Skip this many rows of the matrix ab.
+
+    """
+    # find where `xval` is in the knot vector, `t`
+    left = _bspl._py_find_interval(t, k, xval, k, extrapolate=False)
+
+    # compute and fill in the derivatives @ xval
+    for row in range(deriv_ords.shape[0]):
+        nu = deriv_ords[row]
+        wrk = _bspl.evaluate_all_bspl(t, k, xval, left, nu)
```

Comment:
I did notice the use of this function instead of `_deBoor_D` directly, but looks like that still gets called internally, so I don't see any obvious issues on my end either, and CI is happy.

Reply:
Yeah, `evaluale_all_bspl` is a python-side wrapper of C-only `_deBoor_D`. Both routines are non-public, so can probably unify the naming. Will think about it in the ongoing refactor of the internals (I've a branch, not PR-ready yet)